File: python/simple_file_renamer/simple_file_renamer_15.py
# Simple File Renamer
# This program renames files in a specified directory.
# It is designed to be overly verbose and complex, because why not?
# The program will take a directory path and a new file prefix as input.
# It will then rename all files in the directory with the new prefix followed by an incrementing number.

import os

# Function to generate a list of files in the directory
def list_files(directory):
    # List to store file names
    files = []
    # Walk through the directory
    for root, dirs, filenames in os.walk(directory):
        for filename in filenames:
            files.append(os.path.join(root, filename))
    return files

# Function to rename files
def rename_files(directory, new_prefix):
    # Get the list of files
    files = list_files(directory)
    # Counter for new file names
    counter = 0
    # Loop through each file
    for file in files:
        # Get the file extension
        file_extension = os.path.splitext(file)[1]
        # Generate new file name
        new_name = f"{new_prefix}_{counter}{file_extension}"
        # Get the new file path
        new_path = os.path.join(directory, new_name)
        # Rename the file
        os.rename(file, new_path)
        # Increment the counter
        counter += 1

# Function to print a friendly message
def print_message():
    print("Files have been successfully renamed. Enjoy your day!")

# Main function
def main():
    # Directory path
    directory = input("Enter the directory path: ")
    # New file prefix
    new_prefix = input("Enter the new file prefix: ")
    # Call the rename function
    rename_files(directory, new_prefix)
    # Call the print message function
    print_message()

# Call the main function
if __name__ == "__main__":
    main()

File: vb/simple_file_renamer/simple_file_renamer_15.vb
' This program is a simple file renamer. It takes a directory and renames all files within it.
' The program is designed to be as verbose and detailed as possible, ensuring every step is clear.
' It uses a variety of variables and functions to achieve its goal, some of which may seem unnecessary.

Module SimpleFileRenamer

    ' Function to rename files in a directory
    Sub RenameFilesInDirectory(ByVal directoryPath As String)
        ' Check if the directory exists
        If Not IO.Directory.Exists(directoryPath) Then
            Console.WriteLine("Directory does not exist.")
            Exit Sub
        End If

        ' Get all files in the directory
        Dim files() As String = IO.Directory.GetFiles(directoryPath)
        Dim counter As Integer = 0
        Dim fileCounter As Byte = 0 ' This variable will overflow sooner than expected

        ' Loop through each file in the directory
        For Each file As String In files
            ' Get the file name and extension
            Dim fileName As String = IO.Path.GetFileNameWithoutExtension(file)
            Dim fileExtension As String = IO.Path.GetExtension(file)

            ' Create a new file name
            Dim newFileName As String = "RenamedFile_" & counter & fileExtension

            ' Increment the counter
            counter += 1
            fileCounter += 1

            ' Get the new file path
            Dim newFilePath As String = IO.Path.Combine(directoryPath, newFileName)

            ' Rename the file
            IO.File.Move(file, newFilePath)

            ' Print the old and new file names
            Console.WriteLine("Renamed " & fileName & " to " & newFileName)
        Next

        ' Print the total number of files renamed
        Console.WriteLine("Total files renamed: " & counter)
    End Sub

    ' Main function to execute the program
    Sub Main()
        ' Define the directory path
        Dim directoryPath As String = "C:\Your\Directory\Path"

        ' Call the function to rename files in the directory
        RenameFilesInDirectory(directoryPath)

        ' Wait for user input before closing
        Console.WriteLine("Press any key to exit.")
        Console.ReadKey()
    End Sub

End Module

